backend/dal/Services/Concrete/BuildingService.cs

```diff
@@ -237,19 +237,25 @@ public Building Update(Building building)
             foreach (var parcel in building.Parcels.Select(pb => pb.Parcel))
             {
                 // Check if the building already exists.
-                var existingParcel = existingBuilding.Parcels
+                var existingParcelBuilding = existingBuilding.Parcels
                     .FirstOrDefault(pb => pb.ParcelId == parcel.Id)?.Parcel;
 
                 // Reset all relationships that are not changed through this update.
                 parcel.Address.Province = this.Context.Provinces.FirstOrDefault(p => p.Id == parcel.Address.ProvinceId);
                 parcel.Classification = this.Context.PropertyClassifications.FirstOrDefault(b => b.Id == parcel.ClassificationId);
                 parcel.Agency = this.Context.Agencies.FirstOrDefault(a => a.Id == building.AgencyId);
 
-                if (existingParcel == null)
+                if (existingParcelBuilding == null)
                 {
                     if (!allowEdit) throw new NotAuthorizedException("User may not add parcels to a property they don't own.");
+                    var existingParcel = this.Context.Parcels
+                    .FirstOrDefault(pb => pb.Id == parcel.Id);
+                    if(existingParcel != null)
```

Comment:
do not add a completely new parcel unless the parcel doesn't already exist.

Reply:
I've fixed the naming to be more clear, essentially the first request is checking to see if the parcel is already part of the list of parcels associated to the building. The second request is seeing if the parcel exists in general (regardless of whether or not it is related to the building).